### PR TITLE
Update gen2 links for gen2 fragments

### PR DESCRIPTION
**Pull request by timngyn**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/gen2/quickstart/build-a-backend.mdx [hunk 1]
```diff
@@ -50,7 +50,7 @@ WORK IN THE FRONTEND CODE FILE.)
 
 Using JavaScript or Next.js React Server Components, Middleware, Server
 Actions, or Pages Router? Review how to generate Data clients for those use
-cases: https://docs.amplify.aws/gen2/build-a-backend/data/connect-to-API/
+cases: https://docs.amplify.aws/javascript/build-a-backend/data/connect-to-API/
```
Comment (timngyn) [inline src/fragments/gen2/quickstart/build-a-backend.mdx:53]: For these links should we default to javascript or nextjs?

Comment (reesscot) [reply src/fragments/gen2/quickstart/build-a-backend.mdx:53]: react is our default

File: src/fragments/gen2/quickstart/build-a-backend.mdx [hunk 2]
```diff
@@ -151,7 +151,7 @@ import { defineAuth } from '@aws-amplify/backend';
 /**
  * Define and configure your auth resource
  * When used alongside data, it is automatically configured as an auth provider for data
- * @see https://docs.amplify.aws/gen2/build-a-backend/auth
```
Comment (reesscot) [inline src/fragments/gen2/quickstart/build-a-backend.mdx:154]: Pretty sure this is autogenerated code by the gen2 CLI. Can you doublecheck with the CLI team? We'll want to get these updated in the CLI itself at launch.

Comment (josefaidt) [reply src/fragments/gen2/quickstart/build-a-backend.mdx:154]: to echo @reesscot , it is! https://github.com/aws-amplify/amplify-backend/blob/b0112e34f94b05179addf67c213a5cfce16dc079/packages/create-amplify/templates/basic-auth-data/amplify/auth/resource.ts#L6

File: src/fragments/gen2/quickstart/build-a-backend.mdx (hunk 1 above)
Comment (timngyn) [reply src/fragments/gen2/quickstart/build-a-backend.mdx]: Changed it to react

File: src/fragments/gen2/quickstart/build-a-backend.mdx (hunk 2 above)
Comment (timngyn) [reply src/fragments/gen2/quickstart/build-a-backend.mdx]: Got it, I'll change it back to how it originally was then

**Review by hbuchel — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**